spec/test/stored_queries/test_fulltext_search_flags.py

```diff
@@ -0,0 +1,287 @@
+    """
+    col.import_bulk(
+        [COVID_NO_RANK_STRAIN, COVID_GENUS, COVID_SPECIES, COVID_STRAIN, COV_SPECIES]
+    )
+
+    expected = _add_id(col.name, [COVID_NO_RANK_STRAIN, COVID_SPECIES, COVID_STRAIN])
+
+    _check_queries(re_client, col.name, expected, "Covid")
+
+
+def test_missing_species_or_below_flag_cov(col, re_client):
+    """
+    Test queries against documents that don't have the species_or_below flag at all.
+    """
+    col.import_bulk(
+        [COVID_NO_RANK_STRAIN, COVID_GENUS, COVID_SPECIES, COVID_STRAIN, COV_SPECIES]
+    )
+
+    expected = _add_id(
+        col.name, [COV_SPECIES, COVID_NO_RANK_STRAIN, COVID_SPECIES, COVID_STRAIN]
+    )
+
+    _check_queries(re_client, col.name, expected, "Cov")
+
+
+def test_with_species_or_below_flag_covid(col, re_client):
+    """
+    Test queries against documents that have the new species_or_below flag.
+    """
+    col.import_bulk(
+        [
+            COVID_NO_RANK_SOB_TRUE,
+            COV_SPECIES,
+            COVID_NO_RANK_STRAIN,
+            COVID_GENUS,
+            COVID_CLADE_SOB_TRUE,
+            COVID_SPECIES,
+            COVID_SPECIES_SOB_TRUE,
+            COVID_STRAIN,
+            COVID_GENUS_SOB_FALSE,
+        ]
+    )
+    expected = _add_id(
+        col.name,
+        [
+            COVID_NO_RANK_STRAIN,
+            COVID_SPECIES,
+            COVID_STRAIN,
+            COVID_CLADE_SOB_TRUE,
+            COVID_SPECIES_SOB_TRUE,
+            COVID_NO_RANK_SOB_TRUE,
+        ],
+    )
+    _check_queries(re_client, col.name, expected, "Covid")
+
+
+def test_with_species_or_below_flag_cov(col, re_client):
+    """
+    Test queries against documents that have the new species_or_below.
+    """
+    col.import_bulk(
+        [
+            COVID_NO_RANK_SOB_TRUE,
+            COV_SPECIES,
+            COVID_NO_RANK_STRAIN,
+            COVID_GENUS,
+            COVID_CLADE_SOB_TRUE,
+            COVID_SPECIES,
+            COVID_SPECIES_SOB_TRUE,
+            COVID_STRAIN,
+            COVID_GENUS_SOB_FALSE,
+        ]
+    )
+    expected = _add_id(
+        col.name,
+        [
+            COV_SPECIES,
+            COVID_NO_RANK_STRAIN,
+            COVID_SPECIES,
+            COVID_STRAIN,
+            COVID_CLADE_SOB_TRUE,
+            COVID_SPECIES_SOB_TRUE,
+            COVID_NO_RANK_SOB_TRUE,
+        ],
+    )
+    _check_queries(re_client, col.name, expected, "Cov")
+
+
+def _add_id(col_name, list_o_dicts):
+    return [d | {"_id": f"{col_name}/{d['_key']}"} for d in list_o_dicts]
+
+
+def _check_queries(re_client, col_name, expected_sorted, search_text):
+    res = _run_query(re_client, col_name, _QUERY_NO_SORT, search_text)
+    assert set(_freeze(res)) == set(_freeze(expected_sorted))
+
+    res = _run_query(re_client, col_name, _QUERY_SORT, search_text)
+    assert res == expected_sorted
+
+
+def _freeze(list_o_dicts):
+    return [frozendict(d) for d in list_o_dicts]
```

Comment:
What does `frozendict` do

Reply:
Super simple https://github.com/Marco-Sulla/python-frozendict#introduction

I'm just using it here because the list gets hashed elsewhere when it's put in a `set` so if the `dicts` aren't immutable python will complain

Reply:
You could also do this without importing frozendict ... I'm never sure when a 3rd-party package is good but you probably have confidence in this one

Reply:
> You could also do this without importing frozendict

That be great, but how? There isn't an equivalent to frozendict in the stdlib (which I've always thought is bizarre, but there you go...)

Reply:
I guess if this is just the general purpose solution to hashing dictionaries, this is a great trick I've learned! And totally the appropriate solution.

I was just thinking in this _exact, isolated_ situation you could also deduplicate the list of dicts instead of casting the list to a set, or other more convoluted rabbit hole ideas that could be investigated

Reply:
The reason I'm casting to a set is because the result is unordered, and so just deduplicating the list would cause the test to fail. The test just needs to check that the result has the correct elements, regardless of order

Reply:
Ah. Didn't realize pytest had that limitation! unittest doesn't care. In that case ... `sorted(res)` with a key being a value of the document?

Reply:
I don't think it's pytest related surely? `assertEquals([1, 2], [2, 1])` would fail just as hard as `assert [1, 2] == [2, 1]`

> sorted(res)

Hm, that would work... it just feels wrong, somehow, compared to the set approach, which is closer to the test criteria, which is 'check that the results returned match these expectations, regardless of order`.

I think of think we've spent way too much time over something pretty trivial at this point, so unless you feel strongly about it I'd say we should just leave it as is

Reply:
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual

Yeah, sorry, all good

Reply:
Hmm, neat, although that name doesn't really describe what it does.

Reply:
I assume you can always use unittest assertions in pytest code...